### PR TITLE
Implement typedef and extern type parser

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- introduce `token_dispatch!` for looping over tokens
- refactor `collect_import_spans` and `collect_typedef_spans` to use new macro

## Testing
- `make fmt`
- `make lint`
- `make test`

------
https://chatgpt.com/codex/tasks/task_e_685db5653c088322bf9e501b9bd65850

## Summary by Sourcery

Implement parsing for `typedef` and `extern type` statements alongside existing `import` parsing by introducing a reusable token dispatch macro, refactoring span collection, extending the CST builder to group typedef spans, and enriching the AST with a new `TypeDef` node; bolster coverage with targeted unit tests.

New Features:
- Add support for parsing `typedef` and `extern type` declarations and collect their spans
- Introduce a `TypeDef` AST node with methods to retrieve the defined type’s name and extern flag

Enhancements:
- Introduce `token_dispatch!` macro to unify token scanning across import and typedef span collectors
- Refactor import span collection and CST builder to leverage generic dispatch for both import and typedef spans
- Extend CST construction to wrap typedef spans into `N_TYPE_DEF` nodes and validate sorted spans
- Add helper functions (`skip_tokens_until`, `line_end`, `skip_ws_no_newline`, etc.) for improved single-line parsing and span management

Tests:
- Add comprehensive tests for typedef and extern type parsing, including edge cases and error recovery
- Add a test to verify import error recovery followed by valid import parsing

### CODE REVIEW

**Comment by leynos** [comment]
@coderabbitai review

**Comment by leynos** [comment]
@coderabbitai review

File: src/parser/mod.rs [hunk 1]
```diff
@@ -61,101 +79,284 @@ pub fn parse(src: &str) -> Parsed {
...
+
+    fn handle_typedef(st: &mut State<'_>, span: Span) {
+        let start = span.start;
+        st.i += 1;
+        let end = line_end(st.tokens, st.src, st.i);
+        skip_tokens_until(&mut st.i, st.tokens, end);
+        st.spans.push(start..end);
+    }
+
+    fn handle_extern(st: &mut State<'_>, span: Span) {
```
[the remark this answers is not in the document]
Comment (leynos) [reply src/parser/mod.rs:245]: @coderabbitai new GitHub issue, please

File: src/parser/mod.rs [hunk 2]
```diff
@@ -61,101 +79,284 @@ pub fn parse(src: &str) -> Parsed {
...
+        })
         .ignored();
 
-    let alias = just(SyntaxKind::K_AS)
-        .padded_by(ws.repeated())
-        .ignore_then(ident);
+        let ident = just(SyntaxKind::T_IDENT).ignored().padded_by(ws.repeated());
+
+        let module_path = ident
+            .then(
```
[the remark this answers is not in the document]
Comment (leynos) [reply src/parser/mod.rs:177]: @coderabbitai new GitHub issue, please

File: src/parser/mod.rs [hunk 3]
```diff
@@ -61,101 +79,289 @@ pub fn parse(src: &str) -> Parsed {
...
-    (res.unwrap_or_default(), errors)
+    st.spans
 }
 
-fn build_green_tree(tokens: Vec<(SyntaxKind, Span)>, src: &str, imports: &[Span]) -> GreenNode {
+/// Construct the CST from the token stream and recorded statement spans.
+///
+/// `imports` and `typedefs` contain the ranges of completed statements so that
+/// tokens can be wrapped in the appropriate nodes during tree construction.
+fn build_green_tree(
```
[the remark this answers is not in the document]
Comment (leynos) [reply src/parser/mod.rs:282]: @coderabbitai new GitHub issue, please
